test/hotspot/jtreg/compiler/vectorapi/TestSelectFromTwoVectorOp.java

```diff
@@ -0,0 +1,453 @@
+
+            for (int j = 0; j < SIZE; j++) {
+                bindex[i][j] = byteGen1.next().byteValue();
+                sindex[i][j] = shortGen1.next().shortValue();
+            }
+
+            if (i < dindex.length) {
+              dindex[i] = new double[SIZE];
+              lindex[i] = new long[SIZE];
+
+              random.fill(random.uniformDoubles(0, (double) ((nums[i] * 2) - 1)), dindex[i]);
+              random.fill(random.uniformLongs(0, (long) ((nums[i] * 2) - 1)), lindex[i]);
+            }
+
+            random.fill(random.uniformInts(0, (nums[i] * 2) - 1), iindex[i]);
+            random.fill(random.uniformFloats(0, (float)((nums[i] * 2) - 1)), findex[i]);
+        }
+
+        // Populate the sources
+        Generator<Integer> byteGen = random.uniformInts(Byte.MIN_VALUE, Byte.MAX_VALUE);
+        Generator<Integer> shortGen = random.uniformInts(Short.MIN_VALUE, Short.MAX_VALUE);
+
+        for (int i = 0; i < SIZE; i++) {
+            ba[i] = byteGen.next().byteValue();
+            bb[i] = byteGen.next().byteValue();
+
+            sa[i] = shortGen.next().shortValue();
+            sb[i] = shortGen.next().shortValue();
+        }
+
+        random.fill(random.ints(), ia);
+        random.fill(random.ints(), ib);
+        random.fill(random.floats(), fa);
+        random.fill(random.floats(), fb);
+        random.fill(random.longs(), la);
+        random.fill(random.longs(), lb);
+        random.fill(random.doubles(), da);
+        random.fill(random.doubles(), db);
+    }
+
+    // Test SelectFromTwoVector operation for Bytes
+    @ForceInline
+    public static void ByteSelectFromTwoVectorKernel(VectorSpecies SPECIES, byte[] ba,
+                                                     byte[] bb, byte[] bindex) {
+        for (int i = 0; i < SPECIES.loopBound(ba.length); i += SPECIES.length()) {
+            ByteVector.fromArray(SPECIES, bindex, i)
+                .selectFrom(ByteVector.fromArray(SPECIES, ba, i),
+                            ByteVector.fromArray(SPECIES, bb, i))
+                .intoArray(bres, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_8, ">0"},
+        applyIfCPUFeature = {"asimd", "true"})
+    public static void selectFromTwoVector_Byte64() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_64, ba, bb, bindex[0]);
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_16, ">0"},
+        applyIfCPUFeature = {"asimd", "true"})
+    public static void selectFromTwoVector_Byte128() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_128, ba, bb, bindex[1]);
+    }
+
+    @Test
+    @IR(failOn = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_32},
+        applyIfCPUFeatureAnd = {"asimd", "true", "sve2", "false"},
+        applyIf = {"MaxVectorSize", ">=32"})
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_32, ">0"},
```

Comment:
I know that we currently do not have aarch64 machines with vec_length > 16B and SVE2 enabled but for the sake of completeness I added these tests as well because if we ever do have a 32B or 64B machine with SVE2, we will be able to generate `SelectFromTwoVector` IR and the lowered SVE2 `tbl` instruction.